lib/PuppeteerSharp/Browser.cs

```diff
@@ -295,10 +295,18 @@ public async ValueTask DisposeAsync()
                 targetFilter,
                 isPageTargetCallback);
 
-            initAction?.Invoke(browser);
+            try
+            {
+                initAction?.Invoke(browser);
 
-            await browser.AttachAsync().ConfigureAwait(false);
-            return browser;
+                await browser.AttachAsync().ConfigureAwait(false);
+                return browser;
+            }
+            catch
+            {
+                await browser.DisposeAsync().ConfigureAwait(false);
```

Comment:
if `initAction` or `AttachAsync` throws an exception, we now explicitly dispose `browser`.

Reply:
Good point.